MapboxNavigation/RouteTableViewController.swift

```diff
@@ -68,20 +68,25 @@ class RouteTableViewController: UIViewController {
 extension RouteTableViewController: UITableViewDelegate, UITableViewDataSource {
 
     func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+        return routeController.routeProgress.route.legs.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return "Leg \(section + 1)"
     }
     
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return routeController.routeProgress.currentLeg.steps.count
+        return routeController.routeProgress.route.legs[section].steps.count
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: RouteTableViewCellIdentifier, for: indexPath) as! RouteTableViewCell
-        let leg = routeController.routeProgress.currentLeg
+        let legs = routeController.routeProgress.route.legs
+        
         
-        cell.step = leg.steps[indexPath.row]
+         cell.step = legs[indexPath.section].steps[indexPath.row]
         
-        if routeController.routeProgress.currentLegProgress.stepIndex + 1 > indexPath.row {
+        if  routeController.routeProgress.legIndex + 1 > indexPath.section && routeController.routeProgress.currentLegProgress.stepIndex + 1 > indexPath.row  {
```

Comment:
Actually, this code is incorrect but in a different way than I first thought. This code is equivalent to: `if currentLegIndex >= section && currentLegProgress.stepIndex >= row { dim the cell }`. Now if the user is on step 5 of leg 2, the only dimmed cells are those that meet _both_ the following criteria:

1. the section number is at least 2; and
2. the row number is at least 5

In other words:

* Sections 0 and 1 are completely lit, because they fail criteria (1). 🙅‍♂️
* Section 2 is dimmed up to row 4 and lit beginning with row 5. 👍
* Sections 3, 4, etc. are each dimmed up to row 4 but lit beginning with row 5, because rows 5 and beyond fail criteria (2). 🙅‍♂️